src/lexer/lex.c

```diff
@@ -612,9 +619,48 @@ static const char *ttype_names[] = {
 
 const char *ttype_name(TokenType tt) { return ttype_names[tt]; }
 
+int test_ttype_many_chars() {
+    testing_func_setup();
+
+    tassert(ttype_many_chars("foo") == TT_IDENTIFIER);
+    tassert(ttype_many_chars("struct") == TT_STRUCT);
+    tassert(ttype_many_chars("while") == TT_WHILE);
+
+    return 0;
+}
+
+int test_ttype_one_char() {
+    testing_func_setup();
+
+    // Use ttype_from_string
+    tassert(ttype_one_char('a') == TT_IDENTIFIER);
+    tassert(ttype_one_char('1') == TT_LITERAL);
+
+    tassert(ttype_one_char('+') == TT_PLUS);
+    tassert(ttype_one_char('-') == TT_MINUS);
+    tassert(ttype_one_char('>') == TT_GREATER);
+    tassert(ttype_one_char('~') == TT_BNOT);
+
+    return 0;
+}
+
+int test_ttype_name() {
+    testing_func_setup();
+
+    tassert(strcmp(ttype_name(TT_LITERAL), "literal") == 0);
+    tassert(strcmp(ttype_name(TT_PLUS), "+") == 0);
+    tassert(strcmp(ttype_name(TT_SIZEOF), "sizeof") == 0);
+    tassert(strcmp(ttype_name(TT_WHILE), "while") == 0);
+
```

Comment:
testing this function is great! I am worried that as a lookup table with no information about the ordering of the enum, it is a bit fragile though.

Reply:
That's true. Having this is a good canary in the coal mine for names not working. If the tests break, any code that relies on the function will also break. It's a problem with how naming enums works in C in general so we should be careful with the use of the enum + name function.